nipype/utils/filemanip.py

```diff
@@ -707,34 +707,33 @@ def loadpkl(infile):
     pkl_metadata = None
 
     unpkl = None
-    with indirectory(infile.parent):
-        with SoftFileLock('%s.lock' % infile.name):
-            with pklopen(infile.name, 'rb') as pkl_file:
-                try:  # Look if pkl file contains version file
-                    pkl_metadata_line = pkl_file.readline()
-                    pkl_metadata = json.loads(pkl_metadata_line)
-                except (UnicodeDecodeError, json.JSONDecodeError):
-                    # Could not get version info
-                    pkl_file.seek(0)
-                try:
-                    unpkl = pickle.load(pkl_file)
-                except UnicodeDecodeError:
-                    # Was this pickle created with Python 2.x?
-                    unpkl = pickle.load(pkl_file, fix_imports=True, encoding='utf-8')
-                    fmlogger.info('Successfully loaded pkl in compatibility mode.')
-                # Unpickling problems
-                except Exception as e:
-                    if pkl_metadata and 'version' in pkl_metadata:
-                        from nipype import __version__ as version
-                        if pkl_metadata['version'] != version:
-                            fmlogger.error("""\
+    with SoftFileLock('%s.lock' % infile):
+        with pklopen(str(infile), 'rb') as pkl_file:
+            try:  # Look if pkl file contains version file
+                pkl_metadata_line = pkl_file.readline()
+                pkl_metadata = json.loads(pkl_metadata_line)
+            except (UnicodeDecodeError, json.JSONDecodeError):
+                # Could not get version info
+                pkl_file.seek(0)
+            try:
+                unpkl = pickle.load(pkl_file)
```

Comment:
this still needs to be done in the folder where outputs required to exist are.

Reply:
how about the other `chdir`?